File: experiments/llm-reasoning-cost/generated_code/java/04_error_handling_codex.java
public class SafeDivide {
    public static Result<Integer, String> safe_divide(int numerator, int denominator) {
        if (denominator == 0) {
            return Result.err("division by zero");
        }
        return Result.ok(numerator / denominator);
    }

    public static final class Result<T, E> {
        private final T ok;
        private final E err;
        private final boolean isOk;

        private Result(T ok, E err, boolean isOk) {
            this.ok = ok;
            this.err = err;
            this.isOk = isOk;
        }

        public static <T, E> Result<T, E> ok(T value) {
            return new Result<>(value, null, true);
        }

        public static <T, E> Result<T, E> err(E error) {
            return new Result<>(null, error, false);
        }

        public boolean isOk() {
            return isOk;
        }

        public boolean isErr() {
            return !isOk;
        }

        public T unwrap() {
            if (!isOk) throw new IllegalStateException("Called unwrap on Err");
            return ok;
        }

        public E unwrapErr() {
            if (isOk) throw new IllegalStateException("Called unwrapErr on Ok");
            return err;
        }

        @Override
        public String toString() {
            return isOk ? "Ok(" + ok + ")" : "Err(" + err + ")";
        }
    }

    public static void main(String[] args) {
        System.out.println(safe_divide(10, 2)); // Ok(5)
        System.out.println(safe_divide(7, 3));  // Ok(2)
        System.out.println(safe_divide(5, 0));  // Err(division by zero)
    }
}

File: experiments/llm-reasoning-cost/generated_code/c/04_error_handling_codex.c
#include <stdint.h>
#include <stdbool.h>
#include <stdio.h>

typedef struct {
    bool is_ok;
    int32_t value;
    const char *error;
} ResultInt32;

ResultInt32 safe_divide(int32_t numerator, int32_t denominator) {
    if (denominator == 0) {
        return (ResultInt32){ .is_ok = false, .value = 0, .error = "division by zero" };
    }
    return (ResultInt32){ .is_ok = true, .value = numerator / denominator, .error = NULL };
}